src/utils/notification.utils.ts

```diff
@@ -0,0 +1,153 @@
+ * will ignore that userId and continue to send notifications to the other valid userIds.
+ *
+ * @param {string[]} userIds - The id or ids of the user(s) to receive the notification.
+ * @param {string} title - The title of the notification.
+ * @param {string} body - The content of the notification.
+ * @param {string} notificationType - The type of notification being sent.
+ * @param {number} delay - Optional delay until the notification should be shown to the user. (set as 0 for now)
+ * @param {any} data - Optional data to include with the notification.
+ *
+ * @return {Promise<Notification[]>} - Returns a promise that resolves to an array of notifications.
+ */
+export const sendNotifications = async (
+  userIds: string[],
+  title: string,
+  body: string,
+  notificationType: string,
+  data: any = {},
+  delay: number = 0
+) => {
+  // array of notifications to return
+  let notifications: any[] = [];
+
+  // Exit function if userId is an empty array
+  if (userIds.length === 0) {
+    return notifications;
+  }
+
+  // Javascript will not wait for promises to resolve in a for loop or mapping, so we must collect all the promises
+  const promises = userIds.map(async id => {
+    // check if current user id is valid
+    if (mongoose.Types.ObjectId.isValid(id)) {
+      // add notification to database
+      const notification = await sendNotificationToUser(
+        id,
+        title,
+        body,
+        data,
+        notificationType,
+        delay
+      );
+
+      // add notification to array of notifications to return
+      notifications.push(notification);
+
+      // find user for their notification token
+      const user = await User.findById(id);
+
+      // send notification to user through expo
+      if (user?.notificationToken) {
+        await sendNotificationToExpo(user.notificationToken, title, body, data);
+      }
+    }
+  });
+
+  // Wait for all promise resolutions
+  await Promise.all(promises);
+
+  // return array of notifications
+  return notifications;
+};
+
+/**
+ * Sends a push notification to the specified Expo push token with the given title, body, and data.
+ *
+ * This is a helper function. Function is not exception safe and not to be used directly, instead use sendNotifications.
+ * Expo is a trusted service. If the notification fails to send, it is likely due to an invalid token.
+ * If the token is invalid, the notification will not be sent and no error will be thrown,
+ * it will simply fail silently with the error logged.
+ *
+ * @param expoPushToken The push token of the device to receive the notification.
+ * @param title The title of the notification.
+ * @param body The body text of the notification.
+ * @param data Optional additional data to send along with the notification.
+ */
+export const sendNotificationToExpo = async (
+  expoPushToken: string,
+  title: string,
+  body: string,
+  data: any
+) => {
```

Comment:
use the interface `ExpoNotification` from `Notification.interface.ts` and combine it all into a variable (since we aren't decomposing it) 

```ts
export const sendNotificationToExpo = async (message : ExpoNotification) => {
    try {
        await fetch(..., {...}, body: JSON.stringify(message));
    } catch (error: any) {
        console.log(error?.message);
    }
};
```

Reply:
On it